### PR TITLE
SONARJAVA-4750 Use Java 21 in the CI to build sonar-java

### PR DESCRIPTION
**Pull request by alban-auzeill**

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -102,6 +105,8 @@ ws_scan_task:
 
 qa_os_win_task:
   <<: *WINDOWS_VM_DEFINITION
+  ec2_instance:
+    image: base-windows-jdk21-v*
```
Comment (leveretka) [inline .cirrus.yml:109]: Maybe you should put this change in the `WINDOWS_VM_DEFINITION` on the line 41

File: .cirrus.yml [hunk 2]
```diff
@@ -207,15 +212,19 @@ autoscan_task:
   <<: *ONLY_SONARSOURCE_QA
   eks_container:
     <<: *CONTAINER_DEFINITION
-    image: ${CIRRUS_AWS_ACCOUNT}.dkr.ecr.eu-central-1.amazonaws.com/base:j17-latest
+    dockerfile: .cirrus/Dockerfile.jdk17And21
```
Comment (leveretka) [inline .cirrus.yml:215]: Could you, please, put a comment with TODO, so we can use jdk 21 completely once SQ is compatible

File: its/autoscan/src/test/resources/autoscan/diffs/diff_S6485.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "ruleKey": "S6485",
   "hasTruePositives": false,
-  "falseNegatives": 0,
+  "falseNegatives": 8,
```
Comment (leveretka) [inline its/autoscan/src/test/resources/autoscan/diffs/diff_S6485.json:4]: Looks like many FN, do you know from where this is coming?

**Review by leveretka — COMMENTED** [review]
I had a few questions, maybe we can have a sync?

File: its/autoscan/src/test/resources/autoscan/diffs/diff_S6485.json (hunk 3 above)
Comment (alban-auzeill) [reply its/autoscan/src/test/resources/autoscan/diffs/diff_S6485.json:4]: S6485 only runs if:
```java
  @Override
  public boolean isCompatibleWithJavaVersion(JavaVersion version) {
    return version.isJava19Compatible();
  }
```
So previously, there was no issue at all with or without semantics. Now, we have FN without semantic.

File: .cirrus.yml [hunk 4]
```diff
@@ -207,15 +212,23 @@ autoscan_task:
   <<: *ONLY_SONARSOURCE_QA
   eks_container:
     <<: *CONTAINER_DEFINITION
-    image: ${CIRRUS_AWS_ACCOUNT}.dkr.ecr.eu-central-1.amazonaws.com/base:j17-latest
+    # For now, this autoscan_task need to execute two mvn commands:
+    # * The build of java-checks-test-sources module which requires Java 21.
+    # * The tests using Orchestrator and SonarQube that, for now, fail to work using Java 21
+    # This is why we have a local Dockerfile that provide the 2 versions of Java, 17 and 21.
```
Comment (leveretka) [inline .cirrus.yml:218]: NIce!

File: its/autoscan/src/test/resources/autoscan/diffs/diff_S6485.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "ruleKey": "S6485",
-  "hasTruePositives": false,
+  "hasTruePositives": true,
```
Comment (leveretka) [inline its/autoscan/src/test/resources/autoscan/diffs/diff_S6485.json:3]: Nice!

**Review by leveretka — APPROVED** [review]
LGTM!

**Pull request merged**